crates/biome_glob/src/editorconfig.rs

```diff
@@ -0,0 +1,702 @@
+        assert_eq!(convert_valid_glob("[a/b]{}"), r"\[a/b]\{\}");
+        assert_eq!(convert_valid_glob("{a}"), r"\{a\}");
+        assert_eq!(convert_valid_glob("{a,b"), r"\{a,b");
+
+        assert_eq!(convert_valid_glob("{a,b}"), "{a,b}");
+        assert_eq!(convert_valid_glob("{0,1}"), "{0,1}");
+        assert_eq!(convert_valid_glob("{a,0..1}"), "{a,0..1}");
+
+        assert_eq!(convert_valid_glob("{0..1}"), "{0,1}");
+        assert_eq!(convert_valid_glob("{0..9}"), "{0,1,2,3,4,5,6,7,8,9}");
+        assert_eq!(convert_valid_glob("{+1..+8}"), "{1,2,3,4,5,6,7,8}");
+        assert_eq!(convert_valid_glob("{0..0}"), "{0}");
+
+        assert_eq!(convert_valid_glob("{10..12}"), "{10,11,12}");
+        assert_eq!(convert_valid_glob("{10..10}"), "{10}");
+
+        assert_eq!(convert_valid_glob(r"\{0..0}"), r"\{0..0\}");
+        assert_eq!(convert_valid_glob("{a..b}"), r"\{a..b\}");
+    }
+
+    // Editorconfig glob tests are ported from https://github.com/editorconfig/editorconfig-core-test/tree/master/glob
+
+    #[test]
+    fn test_editorconfig_glob_star() {
+        let mut glob: EditorconfigGlob;
+
+        // matches a single characters
+        glob = "a*e.c".parse().unwrap();
+        assert!(is_matching(&glob, "ace.c"));
+        // matches zero characters
+        assert!(is_matching(&glob, "ae.c"));
+        // matches multiple characters
+        assert!(is_matching(&glob, "abcde.c"));
+        // does not match path separator
+        assert!(!is_matching(&glob, "a/e.c"));
+
+        // star after a slash
+        glob = "Bar/*".parse().unwrap();
+        assert!(is_matching(&glob, "Bar/foo.txt"));
+        // star matches a dot file after slash
+        assert!(is_matching(&glob, "Bar/.editorconfig"));
+        // Slash makes the pattern not match in subdirectories
+        assert!(!is_matching(&glob, "bat/Bar/foo.txt"));
+        // Slash makes the pattern not match dotfiles in subdirectories
+        assert!(!is_matching(&glob, "bat/Bar/.editorconfig"));
+
+        // star matches a dot file
+        glob = "*".parse().unwrap();
+        assert!(is_matching(&glob, ".editorconfig"));
+    }
+
+    #[test]
+    fn test_editorconfig_glob_question() {
+        // matches a single characters
+        let glob: EditorconfigGlob = "som?.c".parse().unwrap();
+        assert!(is_matching(&glob, "some.c"));
+        // does not match zero characters
+        assert!(!is_matching(&glob, "som.c"));
+        // does not match multiple characters
+        assert!(!is_matching(&glob, "something.c"));
+        // does not match slash
+        assert!(!is_matching(&glob, "som/.c"));
+    }
+
+    #[test]
+    fn test_editorconfig_glob_brackets() {
+        // commented tests are tests that should pass and we don't support yet.
+        let mut glob: EditorconfigGlob;
+
+        // close bracket inside
+        //glob = r"[\]ab].g".parse().unwrap();
+        //assert!(is_matching(&glob, "].g"));
```

Comment:
To remove or uncomment?

Reply:
I added a comment to inform devs that commented tests should pass, however we don't support it yet because of limitations.

Reply:
Got it. I would consider adding separate tests, and use the ignore macro with the comment you just used.